montepython/mcmc.py

```diff
@@ -784,6 +784,9 @@ def chain(cosmo, data, command_line):
         # multiplicity of the point and start the loop again
         if get_new_position(
                 data, sigma_eig, U, k, Cholesky, Rotation) is True:
+            # Prints the position even if it is not accepted
+            if command_line.print_cosmo:
+                io_mp.print_vector([data.cosmo_out], 1, 0.0, data)
```

Comment:
If you did not store a file handle before, what you can do is simply to open the file in append mode:
```python
with open(data.cosmo_out_name, 'a') as fid:
    io_mp.print_vector([fid], 1, 0.0, data)
```

Reply:
That's much nicer. I wonder why MontePython doesn't already do this? Maybe it's slow to re-open the each time?